modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -0,0 +1,147 @@
+import {GLTFMaterialParser} from '@luma.gl/experimental';
+import {Model, Geometry, pbr} from '@luma.gl/core';
+import {log} from '@deck.gl/core';
+import {SimpleMeshLayer} from '@deck.gl/mesh-layers';
+
+import vs from './mesh-layer-vertex.glsl';
+import fs from './mesh-layer-fragment.glsl';
+
+function validateGeometryAttributes(attributes, useMeshColors) {
+  const hasColorAttribute = attributes.COLOR_0 || attributes.colors;
+  const useColorAttribute = hasColorAttribute && useMeshColors;
+  if (!useColorAttribute) {
+    attributes.colors = {constant: true, value: new Float32Array([1, 1, 1])};
+  }
+  log.assert(
+    attributes.positions || attributes.POSITION,
+    'no "postions" or "POSITION" attribute in mesh'
+  );
+}
+
+/*
+ * Convert mesh data into geometry
+ * @returns {Geometry} geometry
+ */
+function getGeometry(data, useMeshColors) {
+  if (data.attributes) {
+    validateGeometryAttributes(data.attributes, useMeshColors);
+    if (data instanceof Geometry) {
+      return data;
+    }
+    return new Geometry(data);
+  } else if (data.positions || data.POSITION) {
+    validateGeometryAttributes(data, useMeshColors);
+    return new Geometry({
+      attributes: data
+    });
+  }
+  throw Error('Invalid mesh');
+}
+
+const defaultProps = {
+  // flat or pbr
+  _lighting: {type: 'string', value: 'flat'},
+  // PBR material object. _lighting must be pbr for this to work
+  pbrMaterial: {type: 'object', value: null}
+};
+
+export default class _MeshLayer extends SimpleMeshLayer {
+  getShaders() {
+    const {_lighting} = this.props;
+    const shaders = super.getShaders();
+    const modules = shaders.modules;
+
+    if (_lighting === 'pbr') {
+      modules.push(pbr);
+    }
+
+    return {...shaders, vs, fs};
+  }
+
+  updateState({props, oldProps, changeFlags}) {
+    super.updateState({props, oldProps, changeFlags});
+    if (props._lighting !== oldProps._lighting && props._lighting === 'pbr') {
```

Comment:
Does lighting ever change for a 3D tile? If not this logic can be dropped.

Reply:
Yes. materials came in I3S 1.7. In 1.6 we still use 'flat' lighting.

Reply:
In the future, perhaps we could make the I3S loader generate PBR materials from 1.6 (assuming we can create a PBR material that results in the same or very similar lighting).

This would remove the number of different configurations we need to support during rendering, and is consistent with the general model we have in loaders.gl to upconvert older format versions during loading.